src/smolagents/local_python_executor.py

```diff
@@ -1396,21 +1396,21 @@ def final_answer(value):
     try:
         for node in expression.body:
             result = evaluate_ast(node, state, static_tools, custom_tools, authorized_imports)
-        state["_print_outputs"].value = truncate_content(
-            str(state["_print_outputs"]), max_length=max_print_outputs_length
-        )
+        # Use DEFAULT_MAX_LEN_OUTPUT if max_print_outputs_length is None
```

Comment:
What about moving this to before the `try`
```python
if max_print_outputs_length is None:
    max_print_outputs_length = DEFAULT_MAX_LEN_OUTPUT
```
This way, only 2 lines are needed instead of duplicating in the try clause and except clause.

That said, I thought that if you don't specify a value to `max_print_outputs_length`, it defaults to DEFAULT_MAX_LEN_OUTPUT already. It is only problematic if you explicitly pass `None` to the argument.

Reply:
Good point, maybe a check on None for the argument forcing the default (maybe adding a log) would be even better